### ISSUE TITLE
Tree geometry is too sensitive to long strings

### ISSUE THREAD

**Issue by refob** [PeWu/topola-viewer#34]
My gedcom files contain large locations strings which produce huge node boxes like this:
![location](https://user-images.githubusercontent.com/76838733/103440022-8651ea80-4c42-11eb-8697-dd7cfd560eec.png)

It would help to shorten long strings for nodes and show the full string only in the right detail window.

**Comment by PeWu** [comment]
I agree it would be nice to make the boxes smaller. Do you have a suggestion how to shorten the long strings?

**Comment by endolith** [comment]
For United States locations, shorten to just the state abbreviation

In fact, I can imagine it dynamically showing more or less information as you zoom in and out horizontally, to make the tree wider or narrower, with rules defined that increasingly shorten the text so that it is still maximally useful:

* Cyclone, Keating Township, McKean, Pennsylvania, United States
* Cyclone, Keating Township, McKean, Pennsylvania, US
* Cyclone, Keating Township, McKean, PA, US
* Cyclone, Keating, McKean, PA, US
* Cyclone, McKean, PA, US
* Cyclone, PA, US
* Cyclone, PA
* PA

The totally unrelated https://sigrok.org/wiki/PulseView application has this function, where you can zoom in to a decoded signal and it will show different abbreviations of the content depending on zoom level and what fits:

![image](https://user-images.githubusercontent.com/58611/109429512-0d64da00-79ca-11eb-8daf-f2eef7a7da7b.png)

![image](https://user-images.githubusercontent.com/58611/109429527-1655ab80-79ca-11eb-9328-7eb42778c716.png)

![image](https://user-images.githubusercontent.com/58611/109429533-1e155000-79ca-11eb-9265-e268ab25c221.png)

![image](https://user-images.githubusercontent.com/58611/109429536-279eb800-79ca-11eb-9434-303370020ea6.png)

with the shortening rules defined like:

```py
c1 = 'Left channel' if not self.oldws else 'Right channel'
c2 = 'Left' if not self.oldws else 'Right'
c3 = 'L' if not self.oldws else 'R'
```

**Comment by 370rokas** [comment]
This is also an issue that i've ran into with my tree 😄 

Implemented it in https://github.com/PeWu/topola-viewer/pull/294
